src/Microsoft.OData.Edm/Csdl/CsdlWriter.cs

```diff
@@ -111,6 +110,33 @@ public static bool TryWriteCsdl(IEdmModel model, XmlWriter writer, CsdlTarget ta
             return true;
         }
 
+        /// <summary>
+        /// Outputs a CSDL XML artifact to the provided <see cref="XmlWriter"/>.
+        /// </summary>
+        /// <param name="model">Model to be written.</param>
+        /// <param name="writer">XmlWriter the generated CSDL will be written to.</param>
+        /// <param name="target">Target implementation of the CSDL being generated.</param>
+        /// <param name="writerSettings">The CSDL xml writer settings.</param>
+        /// <param name="errors">Errors that prevented successful serialization, or no errors if serialization was successful. </param>
+        /// <returns>A value indicating whether serialization was successful.</returns>
+        public static bool TryWriteCsdl(IEdmModel model, XmlWriter writer, CsdlTarget target, CsdlXmlWriterSettings writerSettings, out IEnumerable<EdmError> errors)

```

Comment:
The existing `TryWriteCsdl` method could call this new method and pass `new CsdlXmlWriterSettings()` for `writerSettings` parameter to avoid having near replica code in both methods